packages/store/addon/-private/system/ds-model-store.ts

```diff
@@ -32,6 +30,18 @@ type NotificationManager = import('./record-notification-manager').default;
 type DSModel = import('../ts-interfaces/ds-model').DSModel;
 type ShimModelClass = import('./model/shim-model-class').default;
 type DSModelClass = import('@ember-data/model').default;
+type RecordDataClass = typeof import('@ember-data/record-data/-private').RecordData;
+
+// we can't greedily use require as this causes
+// an order of operations issue which gives us undefined
+let _RecordData: RecordDataClass;
+function getRecordData(): RecordDataClass {
+  if (_RecordData !== undefined) {
+    return _RecordData;
+  }
+  _RecordData = HAS_RECORD_DATA_PACKAGE ? require('@ember-data/record-data/-private').RecordData : null;
```

Comment:
Can we move this lazy require directly into `createRecordDataFor`? It would not need the ternary (because that branch is only used when `HAS_RECORD_DATA_PACKAGE` is present)...

Reply:
@rwjblue what is the perf of repeated calls into require? similar cache to our manual one here?

Reply:
ya, I meant more like:

```ts
let _RecordData: undefined | RecordDataClass;

// ...snip...
createRecordDataFor(modelName: string, id: string | null, clientId: string, storeWrapper: RecordDataStoreWrapper) {
  if (HAS_RECORD_DATA_PACKAGE) {
    if (_RecordData === undefined) {
      _RecordData = require('whatever/path');
    }
   
  }
}
```